Zend/zend_constants.c

```diff
@@ -82,11 +82,7 @@ static int clean_module_constant(zval *el, void *arg)
 	zend_constant *c = (zend_constant *)Z_PTR_P(el);
 	int module_number = *(int *)arg;
 
-	if (ZEND_CONSTANT_MODULE_NUMBER(c) == module_number) {
```

Comment:
This function is a callback for zend_hash_apply_with_arguments. The return value is one of the ZEND_HASH_APPLY_* constants defined here: https://github.com/php/php-src/blob/cd9f0a22fc0fd51e86043246bb4e12d39becdcd9/Zend/zend_hash.h#L133

Instead of returning a boolean here, it would probably make more sense to replace the 1 and 0 with ZEND_HASH_APPLY_REMOVE and ZEND_HASH_APPLY_KEEP respectively. The same also applies to a couple of other zend_hash_apply callbacks.